### PR TITLE
Initializes the PartiQLParser (Parser Only)

### PR DESCRIPTION
**Pull request by johnedquinn**
## Description
- Updates `unreleased` section of CHANGELOG
- No backward incompatibility changes
- Introduces an external dependency on ANTLR
- Adds the ANTLR grammar files in `partiql-syntax`
- Adds 2 new metas. See their Java Docs.
- Adds a PartiQLVisitor to convert the ParseTree into a PartiqlAstNode
- The PartiQLParser uses the generated parser and the PartiQLVisitor
- Adds syntax error handlers to continue throwing Parser and Lexer Exceptions whenever ANTLR encounters a syntax error

## More Information
- This PR will not be merged. It is solely to help split up the PR into reviewable chunks. Please see #711 for approvals.
- This PR focuses on introducing the PartiQLParser and ANTLR Grammar

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by am357 — COMMENTED** [review]
Thank you for putting together this change; this significantly simplifies our future Parser changes. I've reviewed most of the code, and did a cursory review of `PartiQLVisitor.kt` and `PartiQLTokens.g4`; submitting my review so far.

**Comment by johnedquinn** [comment]
> Thank you for putting together this change; this significantly simplifies our future Parser changes. I've reviewed most of the code, and did a cursory review of `PartiQLVisitor.kt` and `PartiQLTokens.g4`; submitting my review so far.

Resolved the comments. Thanks for looking at the PR! LMK if you need anything else.

File: partiql-syntax/src/main/antlr/PartiQL.g4 [hunk 1]
```diff
@@ -0,0 +1,602 @@
+
```
Comment (am357) [inline partiql-syntax/src/main/antlr/PartiQL.g4:1]: Nothing major, just curious: do we want to include the Table of operator precedence in the comments?

**Review by am357 — DISMISSED** [review]

File: partiql-syntax/src/main/antlr/PartiQL.g4 (hunk 1 above)
Comment (johnedquinn) [reply partiql-syntax/src/main/antlr/PartiQL.g4:1]: Will do. Good idea.

File: lang/src/org/partiql/lang/syntax/PartiQLParserBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * language governing permissions and limitations under the License.
+ */
+
+package org.partiql.lang.syntax
+
+import com.amazon.ion.IonSystem
+import com.amazon.ion.system.IonSystemBuilder
+import org.partiql.lang.types.CustomType
+
+class PartiQLParserBuilder {
```
Comment (am357) [inline lang/src/org/partiql/lang/syntax/PartiQLParserBuilder.kt:21]: `kdoc` w/ example code?

**Review by am357 — APPROVED** [review]

**Comment by johnedquinn** [comment]
I opted to move `antler-syntax` into an `antlr-grammar` sub-module that also publishes the G4 files as resources. Then, `lang`, also grabs the G4 as resources and publishes them while hiding the implementation of the generated parser.

**Pull request closed without merging**